### ISSUE TITLE
CVE-2023-2976 has not been fixed on f40

### ISSUE THREAD

**Issue by jackchan-x** [fedora-java/javapackages-bootstrap#129]
Hello, a vulnerability CVE-2023-2976 was recently reported in Guava. The upstream community suggests upgrading to version 32.0.1, but the latest version in Fedora is still 31.1 and there is no patch to fix this vulnerability in Fedora. May I ask why Fedora has not fixed this vulnerability?
References: 
- https://github.com/google/guava/issues/2575
- https://github.com/advisories/GHSA-7g45-4rm6-3mm3

**Comment by mizdebsk** [comment]
First, Fedora bugs are tracked in Red Hat Bugzilla at https://bugzilla.redhat.com/.
In case of CVE-2023-2976, the bugs are:
* https://bugzilla.redhat.com/show_bug.cgi?id=2232209
* https://bugzilla.redhat.com/show_bug.cgi?id=2232210

We are planning to fix the issue in Fedora, but it has moderate severity, so it was preempted by more important work.

As for javapackages-bootstrap, the use of guava is very limited and therefore has low severity. It will be fixed eventually by updating Guava to newer version, but it's not a priority for the project.

**Comment by jackchan-x** [comment]
> First, Fedora bugs are tracked in Red Hat Bugzilla at https://bugzilla.redhat.com/. In case of [CVE-2023-2976](https://github.com/advisories/GHSA-7g45-4rm6-3mm3), the bugs are:
> 
> * https://bugzilla.redhat.com/show_bug.cgi?id=2232209
> * https://bugzilla.redhat.com/show_bug.cgi?id=2232210
> 
> We are planning to fix the issue in Fedora, but it has moderate severity, so it was preempted by more important work.
> 
> As for javapackages-bootstrap, the use of guava is very limited and therefore has low severity. It will be fixed eventually by updating Guava to newer version, but it's not a priority for the project.
Thanks.

**Comment by rikenm1** [comment]
Hi Fedora team, 
We are using a very old jpb-bootstrap version in Azure Linux i.e 1.5.0. I was updating the source to guava source 2.1.3 for this exact CVE. 
I update the guava.xml and guava.properties in projects. 1.5.0 did not have jurand hence I had to still use the older guava.properties from 1.5.0 .
The build fails at the guava compilation stage. Ant stage went fine. There is no log or anything to debug this error. 
```
time="2024-03-22T15:00:01Z" level=debug msg="guava: compiler"
time="2024-03-22T15:00:02Z" level=debug msg="Failed to execute tool compiler on module guava: Compilation failed"
time="2024-03-22T15:00:02Z" level=debug msg="BUILD FAILURE"

```
As it failed during the compile time, It may be jsr-305 version incompatibility with guava which is causing it.  I have updated other sources which did not have this issue. Any input?
